src/server/api/routers/board.ts

```diff
@@ -86,4 +88,96 @@ export const boardRouter = createTRPCRouter({
       const targetPath = `data/configs/${input.boardName}.json`;
       fs.writeFileSync(targetPath, JSON.stringify(newConfig, null, 2), 'utf8');
     }),
+  renameBoard: protectedProcedure
+    .input(z.object({
+      oldName: z.string(),
+      newName: z.string().min(1),
+    }))
+    .mutation(async ({ input }) => {
+      if (input.oldName === 'default') {
+        Consola.error(`Attempted to rename default configuration. Aborted deletion.`);
+        throw new TRPCError({
+          code: 'CONFLICT',
+          message: 'Cannot rename default board',
+        });
+      }
+
+      if (!configExists(input.oldName)) {
+        Consola.error(`Specified configuration ${input.oldName} does not exist on file system`);
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: 'Board not found',
+        });
+      }
+
+      if (configExists(input.newName)) {
+        Consola.error(`Target name of rename conflicts with existing board`);
+        throw new TRPCError({
+          code: 'CONFLICT',
+          message: 'Board conflicts with existing board',
+        });
+      }
+
+      const config = getConfig(input.oldName);
+      config.configProperties.name = input.newName;
+      writeConfig(config);
+      Consola.info(`Deleting ${input.oldName} from the file system`);
+      const targetPath = `data/configs/${input.oldName}.json`;
+      fs.unlinkSync(targetPath);
+      Consola.info(`Deleted ${input.oldName} from file system`);
+    }),
+  duplicateBoard: protectedProcedure
+    .input(z.object({
+      boardName: z.string(),
+    }))
+    .mutation(async ({ input }) => {
+      if (!configExists(input.boardName)) {
+        Consola.error(`Tried to duplicate ${input.boardName} but this configuration does not exist.`);
+        throw new TRPCError({
+          code: 'NOT_FOUND',
+          message: 'Board not found',
+        });
+      }
+
+      const targetName = attemptGenerateDuplicateName(input.boardName, 10);
+
+      Consola.info(`Target duplication name ${targetName} does not exist`);
+
+      const config = getConfig(input.boardName);
+      config.configProperties.name = targetName;
+      writeConfig(config);
+
+      Consola.info(`Wrote config to name '${targetName}'`)
+    }),
 });
+
+const duplicationName = /^(\w+)\s{1}\(([0-9]+)\)$/;
+
+const attemptGenerateDuplicateName = (baseName: string, maxAttempts: number) => {
```

Comment:
When I duplicate in the following order it will fail:
- duplicate `board` --> `board (2)`
- duplicate `board` again --> Error because `board (2)` already exists

Reply:
No, it shouldn't. It should continue at ``board (3)``. Did you actually test locally? It works on my machine (ironic, isn't it)